### PR TITLE
OOP Advanced

### PR DESCRIPTION
**Pull request by VladSlob**

### CODE REVIEW

**Review by Rommelua — CHANGES_REQUESTED** [review]
Java Cl checks failed. Before making a Pull Request (or a new commit to an already existing PR), you need to write the mvn clean package command in the terminal and make sure that all checks pass. After that, the mentor will be able to review the homework

image

File: src/main/java/core/basesyntax/Calculable.java [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package core.basesyntax;
+
+public interface Calculable {
```
Comment (kerrrusha) [inline src/main/java/core/basesyntax/Calculable.java:3]:
```suggestion
public interface AreaCalculator {
```

File: src/main/java/core/basesyntax/Calculable.java [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+    
+    double getArea();
```
Comment (kerrrusha) [inline src/main/java/core/basesyntax/Calculable.java:5]: if interface calculates - lets keep that concept
```suggestion
    double calculateArea();
```

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class ColorSupplier {
+    private static final String[] colors = {"blue", "red", "green", "yellow", "black"};
```
Comment (kerrrusha) [inline src/main/java/core/basesyntax/ColorSupplier.java:6]: create enum for that
```suggestion
```

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+    public String getRandomColor() {
+        return colors[new Random().nextInt(colors.length)];
```
Comment (kerrrusha) [inline src/main/java/core/basesyntax/ColorSupplier.java:9]: random should be private final field
```suggestion
        return colors[random.nextInt(colors.length)];
```

File: src/main/java/core/basesyntax/Drawable.java [hunk 5]
```diff
@@ -0,0 +1,6 @@
+package core.basesyntax;
+
+public interface Drawable {
+
```
Comment (kerrrusha) [inline src/main/java/core/basesyntax/Drawable.java:4]:
```suggestion
```

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX_PARAMETER = 10;
+    private Random random = new Random();
+
+    public Shape getRandomFigure() {
+        int figureType = random.nextInt(5);
```
Comment (kerrrusha) [inline src/main/java/core/basesyntax/FigureSupplier.java:10]: 5 is a magic number

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+                    color, 
+                    random.nextInt(MAX_PARAMETER) + 1, 
+                    random.nextInt(MAX_PARAMETER) + 1
+                );
+            case 3:
+                return new IsoscelesTrapezoid(
+                    color, 
+                    random.nextInt(MAX_PARAMETER) + 1, 
+                    random.nextInt(MAX_PARAMETER) + 1, 
+                    random.nextInt(MAX_PARAMETER) + 1
```
Comment (kerrrusha) [inline src/main/java/core/basesyntax/FigureSupplier.java:29]: `random.nextInt(MAX_PARAMETER) + 1` repeats, create method for that and reuse it

**Review by kerrrusha — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/ColorSupplier.java (hunk 3 above)
Comment (kerrrusha) [reply src/main/java/core/basesyntax/ColorSupplier.java:6]: not fixed

**Review by kerrrusha — CHANGES_REQUESTED** [review]

**Pull request closed without merging**